Insights/Usecases/Visit.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Visit.swift
+//  Insights
+//
+//  Created by Vladislav Fitc on 26/11/2018.
+//  Copyright © 2018 Algolia. All rights reserved.
+//
+
+import Foundation
+
+/// Provides convenient functions for tracking events which can be used for search personalization.
+///
+
+class Visit: NSObject, AnalyticsUsecase, CustomEventTrackable {
+    
+    var eventProcessor: EventProcessable
+    var logger: Logger
+    var userToken: String?
+    
+    init(eventProcessor: EventProcessable,
+         logger: Logger,
+         userToken: String? = .none) {
+        self.eventProcessor = eventProcessor
+        self.logger = logger
+        self.userToken = userToken
+    }
+    
+    func view(eventName: String,
+              indexName: String,
+              userToken: String? = .none,
+              timestamp: Int64 = Date().millisecondsSince1970,
+              objectIDs: [String]) {
+        do {
+            
+            let event = try View(name: eventName,
+                                 indexName: indexName,
+                                 userToken: effectiveUserToken(withEventUserToken: userToken),
+                                 timestamp: timestamp,
+                                 queryID: .none,
+                                 objectIDsOrFilters: .objectIDs(objectIDs))
+            eventProcessor.process(event)
+            
+        } catch let error {
+            logger.debug(message: error.localizedDescription)
+        }
+    }
+    
+    func view(eventName: String,
+              indexName: String,
+              userToken: String? = .none,
+              timestamp: Int64 = Date().millisecondsSince1970,
```

Comment:
Based on the last week problems and conversation, we should not default the timestamp. The server accepts optional timestamp

Reply:
Same for all the other methods